rivertype/job_row.go

```diff
@@ -0,0 +1,116 @@
+// Package rivertype stores some of the lowest level River primitives so they
+// can be shared amongst a number of packages including the top-level river
+// package, database drivers, and internal utilities.
+package rivertype
+
+import (
+	"time"
+)
+
+// JobRow contains the properties of a job that are persisted to the database.
+// Use of `Job[T]` will generally be preferred in user-facing code like worker
+// interfaces.
+type JobRow struct {
+	// ID of the job. Generated as part of a Postgres sequence and generally
+	// ascending in nature, but there may be gaps in it as transactions roll
+	// back.
+	ID int64
+
+	// Attempt is the attempt number of the job. Jobs are inserted at 0, the
+	// number is incremented to 1 the first time work its worked, and may
+	// increment further if it's either snoozed or errors.
+	Attempt int
+
+	// AttemptedAt is the time that the job was last worked. Starts out as `nil`
+	// on a new insert.
+	AttemptedAt *time.Time
+
+	// AttemptedBy is the set of worker IDs that have worked this job. A worker
+	// ID differs between different programs, but is shared by all executors
+	// within any given one. (i.e. Different Go processes have different IDs,
+	// but IDs are shared within any given process.) A process generates a new
+	// ULID (an ordered UUID) worker ID when it starts up.
+	AttemptedBy []string
```

Comment:
Post launch we are probably going to want to think more about the worker ID thing and maybe offer a way for users to set this per client to match an internal identifier they already have for the node.

Reply:
Yeah +1. I was also thinking the other day that we might want to try and default to something a little more meaningful — even a hostname or something would be more human-friendly, but yeah, it should be configurable.